### ISSUE TITLE
API errors - none critical 

### ISSUE THREAD

**Issue by fate8383** [thoukydides/homebridge-homeconnect#175]
### Description of Issue

Below logs sums up i think, i am not sure how to describe it, on reboot or hb reboot i am getting these.

[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Door closed
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Inactive (Inactive)
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Connected
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Inactive (Inactive)
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Supports Sabbath Mode
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] Unexpected fields in Home Connect API response:
[11/10/2023, 11:59:11 AM] [HomeConnect] GET /api/homeappliances/SIEMENS-HN....../programs
[11/10/2023, 11:59:11 AM] [HomeConnect] ├─ response.data.active is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] └─ response.data.selected is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Sabbath Mode off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Sabbath Mode off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Can be placed in standby
[11/10/2023, 11:59:11 AM] [HomeConnect] Unexpected fields in Home Connect API response:
[11/10/2023, 11:59:11 AM] [HomeConnect] GET /api/homeappliances/SIEMENS-HN..../programs/available
[11/10/2023, 11:59:11 AM] [HomeConnect] ├─ response.data.active is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] └─ response.data.selected is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] 1 of 14 programs are currently unavailable
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Missing options for 1 programs
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] To update the program options for this appliance:
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Enable remote control on the appliance
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Avoid operating the appliance locally
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Ensure that no program is active
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Fill any consumables that may be required
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Either restart Homebridge or invoke the appliance's Identify routine

### Expected Behaviour

n/a

### Steps to Reproduce

reboot server or hb

### Plugin Version

latest stable

### Environment

- OS: raspy latest stable
- Node: v20 latest stable
- Homebridge: latest stable

### Home Connect Appliance(s)

SIEMENS-HN678G4S6

### HomeKit App(s)

n/a

### Diagnostic Checks

- [X] Home Connect app is able to control the appliance with Wi-Fi disabled.
- [X] Home Connect app shows three green lines in the Network section of the appliance Settings.
- [X] Unofficial [Home Connect Server Status](https://homeconnect.thouky.co.uk) shows servers up and no recent errors.
- [X] Searched for [previously reported issues](https://github.com/thoukydides/homebridge-homeconnect/issues?q=is%3Aissue) and no similar ones found.
- [X] Checked the [Home Connect API errors](https://api-docs.home-connect.com/general?#api-errors) and [authorisation errors](https://api-docs.home-connect.com/authorization?#authorization-errors) documentation.

### Log File

```shell
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Door closed
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Inactive (Inactive)
[11/10/2023, 11:59:10 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Connected
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Inactive (Inactive)
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Supports Sabbath Mode
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] Unexpected fields in Home Connect API response:
[11/10/2023, 11:59:11 AM] [HomeConnect] GET /api/homeappliances/SIEMENS-HN..../programs
[11/10/2023, 11:59:11 AM] [HomeConnect] ├─ response.data.active is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] └─ response.data.selected is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Sabbath Mode off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Sabbath Mode off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Off
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Can be placed in standby
[11/10/2023, 11:59:11 AM] [HomeConnect] Unexpected fields in Home Connect API response:
[11/10/2023, 11:59:11 AM] [HomeConnect] GET /api/homeappliances/SIEMENS-HN.../programs/available
[11/10/2023, 11:59:11 AM] [HomeConnect] ├─ response.data.active is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] └─ response.data.selected is extraneous
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] 1 of 14 programs are currently unavailable
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] Missing options for 1 programs
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven] To update the program options for this appliance:
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Enable remote control on the appliance
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Avoid operating the appliance locally
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Ensure that no program is active
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Fill any consumables that may be required
[11/10/2023, 11:59:11 AM] [HomeConnect] [Oven]   - Either restart Homebridge or invoke the appliance's Identify routine
```

### Configuration

```json
just one device which is oven

        {
            "clientid": "....",
            "simulator": false,
            "language": {
                "api": "en-US"
            },
            "SIEMENS-HN....": {
                "addprograms": "none",
                "programs": []
            },
            "platform": "HomeConnect"
        },
```

**Comment by thoukydides** [comment]
Thank you for reporting this issue. I can reproduce it on my own setup. It looks like the `/programs/available` API method is now returning addition information. The Home Connect API documentation has not been updated to describe the extra information, but it is reasonably self-explanatory.

As you say, this does not affect the plugin operation - the log is just warning that additional unexpected information is being returned by the API. However, I will update the plugin to expect the extra fields, and (probably) make use of them when provided.

**Comment by thoukydides** [comment]
Fixed in v0.30.0.
